### ISSUE TITLE
Fermionic AD silently hangs in _optimize_gs_ad_2site (no gradient steps complete)

### ISSUE THREAD

**Issue by yingjerkao** [tenax-lab/tenax#565]
## Summary

\`_optimize_gs_ad_2site\` in \`src/tenax/algorithms/ipeps_optimize.py\` hangs at the first gradient step when given FermionParity-symmetric site tensors, even in the minimal-cost configuration (D=3, χ=8, 3 AD steps). Process consumes 99% CPU for 30+ min with **zero verbose output after the initial implicit-AD warning**, then was killed. No traceback, no convergence message, no JIT-failure error — silent hang.

For comparison, an equivalent 2-site **bosonic** Heisenberg AD at the same (D=3, χ=8) completes a gradient step in seconds.

## Reproducer

Branch: \`origin/main\` at \`a4bedc6\` (post #560 + #561 + #562 + #564).

Self-contained script (~120 lines, attached as the canonical reproducer):

\`\`\`python
# examples/bench_tV_2site_ad.py
# 2-site bipartite fermionic spinless t-V at PH-symmetric μ=2V.
# Builds FermionParity SymmetricTensor A, B with canonical virtual charges
# [i%2 for i in range(D)] and the standard 2-site t-V + chemical-potential
# bond gate (SymmetricTensor, FermionParity, 4 legs).
\`\`\`

Invocation that hangs:

\`\`\`bash
JAX_PLATFORMS=cpu uv run python -u examples/bench_tV_2site_ad.py
# Config: D=3, χ=8, gs_num_steps=3, gs_verbose=True, gs_log_interval=1, gs_c4v=False
\`\`\`

I'll attach the script to this issue or upload as a PR.

## Observations

1. The script's pre-AD prints fire correctly (header + \"# Starting V=2.0 ...\" line).
2. The implicit-AD path warning at \`ipeps_optimize.py:2343\` fires (\"2-site AD with gs_c4v=False uses the implicit-AD path...\").
3. After the warning, **no further output for 30 min at 99% CPU**, despite \`gs_verbose=True\` and \`gs_log_interval=1\` (which should print per-step energy).
4. \`ctm_max_iter=20\`, \`ctm_conv_tol=1e-4\`, \`lr=5e-3\` — all conservative.

## Likely causes

- The implicit-AD path (\`gs_implicit_ad=True\` by default) uses GMRES through the CTM fixed-point Jacobian. If the CTM doesn't converge (or converges to a state where the Jacobian is rank-deficient/ill-conditioned for fermionic input), GMRES may iterate to its max with no progress markers.
- The fermionic energy path goes through \`compute_energy_ctm_tensor_2site\` which works fine in eager mode (post-#562) — but the *traced/AD* version was never benchmarked on FermionParity input.
- 2-site bipartite AD shell logic in \`_optimize_gs_ad_2site\` may convert tensors through a code path that loses fermionic information silently.

## What works (cross-checks)

| Test | Path | Status |
|---|---|---|
| 2-site SU on FermionParity D=3 | \`_simple_update_2site_horizontal/vertical_tensor\` | ✅ Works, ~30s for 200 steps |
| Native split-aware CTM energy | \`compute_energy_split_ctm_tensor*\` | ✅ Works (#392 / PR #562 validated) |
| Standard double-layer CTM energy | \`compute_energy_ctm_tensor_2site\` (eager) | ✅ Works (used in bench above) |
| Density via 1-site RDM | \`_rdm_1site_tensor\` | ✅ Works |
| Bond-layout preservation at D>2 | (post #560 + #564) | ✅ Works |

So the algebra is verifiably sound — the gap is specifically in the AD orchestration on fermionic input.

## Suggested investigation paths

1. **Add a fermionic 2-site AD regression test** at D=2, χ=4, 1 step, with a hard wall-clock timeout (e.g. 60 s). Anything slower is failure. Even D=2 will hang if the path is broken.
2. **Bisect the AD shell**: run the energy function alone (eager), then through \`jax.value_and_grad\`, then through \`_optimize_gs_ad_tensor_2site\` to find which layer first stalls.
3. **Try explicit AD** (\`gs_implicit_ad=False\`) — bypasses GMRES via unrolled CTM. If explicit works and implicit hangs, the issue is in the implicit-diff fixed-point machinery (variPEPS-style backward), not the forward energy.
4. **Check whether \`AB_init\` Tensor protocol path** preserves FermionParity through any \`_wrap_as_dense_tensor\` / \`todense\` step. The 2-site AD orchestrator has a U(1)-default fast-path that may be the entry point.

## Why this matters

Tenax has shipped fermionic SU + CTM + energy via PRs #557, #560, #561, #562, #564 (this week). The remaining structural gap to call the fermionic stack \"production-ready for symmetry-broken phases\" is AD optimization — SU alone cannot find broken-symmetry GS at D≥3 for the spinless t-V model (verified empirically: SU at D=3 falls into the bond-decoupled symmetric attractor regardless of init or chemical-potential detuning). AD is the only available path for canonical fermionic benchmarks (Corboz Hubbard U=8 half-fill, etc.).

## Related

- #392 — fermionic split-CTM energy (closed by #562; the native χ²·D⁴ path validates the energy half).
- #563 — 2-site SU bond layout at D>2 (closed by #564).
- #511 — chi-ramp ghost-minimum issue (mentioned in the implicit-AD warning).
- variPEPS reference for fermionic implicit AD: arXiv 2308.12358 §2.8.
